game_engine.hpp

```diff
@@ -233,7 +233,7 @@ class game_engine
 
     CONSTEXPR void handle_input(KeyboardInput input)
     {
-        auto move_cursor_relative = [=](int x, int y) constexpr
```

Comment:
Apparently, = will not capture this in C++20. So GCC raises a preventive warning.